libs/journey-bundles/transactions/src/lib/transactions.bundle.ts

```diff
@@ -3,79 +3,60 @@ import {
+const extensionsConfig: Partial<TransactionsJourneyExtensionsConfig> = {
+  transactionItemAdditionalDetails: TransactionItemAdditionalDetailsComponent,
+};
 
-  getJourneyConfig() {
-    return {
-      pageSize: 10,
-      slimMode: this.#slimMode,
-    };
-  }
-}
+// Journey configuration
+const journeyConfig = {
+  pageSize: 10,
+  slimMode: false,
+};
 
-// The actual routes that will be lazy-loaded
-const routes: Routes = transactionsJourney(
-  // Journey configuration - using a factory function that will be called at runtime
-  withConfig({
-    pageSize: 10,
-    slimMode: false, // Default value, will be updated by the module
-  }),
-  // Communication service configuration
+// Providers needed by the journey's internal services
+// Note: TransactionsJourneyConfiguration is an @Injectable class used by internal services
+// (TransactionsHttpService, TransactionsRouteTitleResolverService), which is different from
+// the TRANSACTIONS_JOURNEY_CONFIG InjectionToken provided by withConfig().
+const journeyProviders = [
+  {
+    provide: TransactionsJourneyConfiguration,
+    useValue: journeyConfig,
+  },
+  TransactionsRouteTitleResolverService,
+  {
+    provide: TRANSACTION_EXTENSIONS_CONFIG,
+    useValue: extensionsConfig,
+  },
+];
+
+// Build the journey routes with providers attached
+const journeyRoutes: Routes = transactionsJourney(
+  withConfig(journeyConfig),
   withCommunicationService(JourneyCommunicationService),
-  // Extensions configuration
-  withExtensions({
-    transactionItemAdditionalDetails: TransactionItemAdditionalDetailsComponent,
-  })
+  withExtensions(extensionsConfig)
 );
 
-@NgModule({
-  imports: [RouterModule.forChild(routes)],
-  providers: [
-    TransactionsConfigService,
-    TransactionsRouteTitleResolverService,
-    {
-      provide: TransactionsJourneyConfiguration,
-      useFactory: (configService: TransactionsConfigService) => {
-        const config = configService.getJourneyConfig();
-        const journeyConfig = new TransactionsJourneyConfiguration();
-        journeyConfig.pageSize = config.pageSize;
-        journeyConfig.slimMode = config.slimMode;
-        return journeyConfig;
-      },
-      deps: [TransactionsConfigService],
-    },
-    {
-      provide: TRANSACTIONS_JOURNEY_CONFIG,
-      useFactory: (configService: TransactionsConfigService) => {
-        return configService.getJourneyConfig();
-      },
-      deps: [TransactionsConfigService],
-    },
-  ],
-})
-export class TransactionsModule {
-  constructor(configService: TransactionsConfigService) {
-    // Get configuration at runtime
-    const config = configService.getJourneyConfig();
-
-    // You could update route components here if needed
-  }
-}
+// Create wrapper route that includes the required providers
+// This ensures providers are available when the journey is lazy-loaded
+const routes: Routes = [
+  {
+    path: '',
+    providers: journeyProviders,
+    children: journeyRoutes,
+  },
```

Comment:
We can use `withProviders` from foundation-ang inside the `transactionsJourney(...)` call above to avoid needing this wrapper route (this is what `withProviders` is designed for)